FF123 updates for permission publickey-credentials-create (#32135)
mdn/content: files/en-us/web/api/credentialscontainer/create/index.md

### Exceptions

- `NotAllowedError` {{domxref("DOMException")}}
  - : Possible causes include:
    - Usage was blocked by a {{HTTPHeader("Permissions-Policy/publickey-credentials-create","publickey-credentials-create")}} [Permissions Policy](/en-US/docs/Web/HTTP/Permissions_Policy).

>     - The function is called cross-origin when the {{HTTPHeader("Permissions-Policy/publickey-credentials-create","publickey-credentials-create")}} policy is set by [`allow=` on an iframe](/en-US/docs/Web/HTTP/Headers/Permissions-Policy#iframes) (and permitted), but the frame does not also have {{glossary("Transient activation")}}.

Reviewer: I found this a bit hard to understand. For one thing I thought having "is set by `allow=` on an iframe" but linking to /en-US/docs/Web/HTTP/Headers/Permissions-Policy#iframes - which is billed as an example* - is odd. Why not link to https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe#allow here?  

\* pet peeve of mine is when examples introduce new normative information instead of being illustrative

But that apart it reads like there are three conditions here that give rise to the exception:
1. cross-origin call
2. `publickey-credentials-create` is set on the iframe (and not sure what "(and permitted)" means)
3. the iframe does not have transient activation

But (2) doesn't look like a reason for _failure_, it looks like a prerequisite for _success_, no?

So could we put it like this:

```suggestion
    - The function is called cross-origin but the iframe's [`allow`](/en-US/docs/Web/HTML/Element/iframe#allow) attribute does not set the appropriate {{HTTPHeader("Permissions-Policy/publickey-credentials-create","publickey-credentials-create")}} policy.
    - The function is called cross-origin and the iframe does not have {{glossary("transient activation")}}.
```

...i.e. list two failure conditions? Is that even correct? If it is, this is easier me to to understand at any rate.

Author: @wbamberg 
With respect to the conditions, I think you have it. 
To make a cross origin call you need an iframe `allow` that specifies an appropriate policy to allow the gated feature to be used in the origin of the loaded file. The iframe also needs transient activation - i.e. it will fail unless both conditions are met. However it will not fail if the iframe is same origin even without transient activation. 

I'm going to merge this, but ...

I linked to https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Permissions-Policy#iframes because that actually explains how to use permission policy and iframes, while https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe#allow says what it is for and then links to the top level Permissions-Policy doc. 

So essentially to find the information needed to actually use this feature now requires two clicks and some luck.
  
We could provide the more direct link I specify from `allow`, and perhaps also have an example here? 
OR we could create a proper standalone doc for iframe allow attribute - is there a model to copy for attributes? I see https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/readonly and similar but these all appear to be globals. Thoughts?

Reviewer: > I linked to https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Permissions-Policy#iframes because that actually explains how to use permission policy and iframes, while https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe#allow says what it is for and then links to the top level Permissions-Policy doc.

Yeah, when I actually looked at the link targets I saw that too, which I suppose is why I was whining about it being in the examples section. It is good content. So I suppose minimally we should deep link there from https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe#allow.

> OR we could create a proper standalone doc for iframe allow attribute - is there a model to copy for attributes? I see https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/readonly and similar but these all appear to be globals. Thoughts?

AIUI they're not globals. For `readonly`, for example:

> The readonly attribute is supported by [text](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/text), [search](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/search), [url](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/url), [tel](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/tel), [email](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/email), [password](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/password), [date](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/date), [month](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/month), [week](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/week), [time](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/time), [datetime-local](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/datetime-local), and [number](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number) [<input>](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input) types and the [<textarea>](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea) form control elements. If present on any of these input types and elements, the [:read-only](https://developer.mozilla.org/en-US/docs/Web/CSS/:read-only) pseudo class will match. If the attribute is not included, the [:read-write](https://developer.mozilla.org/en-US/docs/Web/CSS/:read-write) pseudo class will match.

What do you think about:
- in this PR, adding the deep-link from https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe#allow to https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Permissions-Policy#iframes
- filing an issue for a standalone `allow` page
?

Author: Good plan:
- Added link in https://github.com/mdn/content/pull/32135/commits/fad947eb252a7b02bf5ffe20046b52fda719d3a3
- I created the issue here https://github.com/mdn/content/issues/32188 - I don't expect I will necessarily do the work.
